pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -452,4 +522,60 @@ static void _session_to_ep(const session_t *session, sock_udp_ep_t *ep)
     memcpy(&ep->addr.ipv6, &session->addr, sizeof(ipv6_addr_t));
 }
 
+#ifdef SOCK_HAS_ASYNC
+void _udp_cb(sock_udp_t *udp_sock, sock_async_flags_t flags, void *ctx)
+{
+    if (flags & SOCK_ASYNC_MSG_RECV) {
+        sock_dtls_t *sock = ctx;
+        sock_dtls_session_t remote;
+        void *data;
+        void *data_ctx;
+
+        ssize_t res = sock_udp_recv_buf(udp_sock, &data, &data_ctx, 0,
+                                        &remote.ep);
+        if (res <= 0) {
+            DEBUG("sock_dtls: error receiving UDP packet: %d\n", (int)res);
+            return;
+        }
+
+        if (sock->buf_ctx != NULL) {
+            DEBUG("sock_dtls: unable to store buffer asynchronously\n");
+            _check_more_chunks(udp_sock, &data, &data_ctx, &remote.ep);
+            return;
+        }
```

Comment:
I had to comment this out in my [patch](https://gist.github.com/pokgak/195b61e15eadffe2a211cf3866eb7a55) to get it working. From https://github.com/RIOT-OS/RIOT/pull/12907#discussion_r404143233 I understand that this check is for the chunked UDP payload support which is still not in tinydtls. Can you explain how checking for `sock->buf_ctx` can check for chunked UDP payload?

Reply:
Also why check for `sock->buf_ctx` when in line 532 and 534 you're using `data_ctx`?